python/lsst/meas/deblender/baselineUtils.cc

```diff
@@ -0,0 +1,123 @@
+    cls.def_static("medianFilter", &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::medianFilter,
+                   "img"_a, "outimg"_a, "halfsize"_a);
+    cls.def_static("makeMonotonic", &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::makeMonotonic,
+                   "img"_a, "pk"_a);
+    // apportionFlux expects an empty vector containing HeavyFootprint pointers that is modified
+    // in the function. But when a list is passed to pybind11 in place of the vector,
+    // the changes are not passed back to python. So instead we create the vector in this lambda and
+    // include it in the return value.
+    cls.def_static("apportionFlux",
+                   [](MaskedImageT const& img, lsst::afw::detection::Footprint const& foot,
+                      std::vector<typename std::shared_ptr<lsst::afw::image::Image<ImagePixelT>>> templates,
+                      std::vector<std::shared_ptr<lsst::afw::detection::Footprint>> templ_footprints,
+                      ImagePtrT templ_sum, std::vector<bool> const& ispsf, std::vector<int> const& pkx,
+                      std::vector<int> const& pky, int strayFluxOptions, double clipStrayFluxFraction) {
+                       using HeavyFootprintPtrList =
+                               std::vector<std::shared_ptr<typename lsst::afw::detection::HeavyFootprint<
+                                       ImagePixelT, MaskPixelT, VariancePixelT>>>;
+
+                       std::vector<typename std::shared_ptr<
+                               lsst::afw::image::MaskedImage<ImagePixelT, MaskPixelT, VariancePixelT>>>
+                               result;
+                       HeavyFootprintPtrList strays;
+                       result = BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::apportionFlux(
+                               img, foot, templates, templ_footprints, templ_sum, ispsf, pkx, pky, strays,
+                               strayFluxOptions, clipStrayFluxFraction);
+
+                       return py::make_tuple(result, strays);
+                   });
+    cls.def_static("hasSignificantFluxAtEdge",
+                   &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::hasSignificantFluxAtEdge, "img"_a,
+                   "sfoot"_a, "thresh"_a);
+    cls.def_static("getSignificantEdgePixels",
+                   &BaselineUtils<ImagePixelT, MaskPixelT, VariancePixelT>::getSignificantEdgePixels, "img"_a,
+                   "sfoot"_a, "thresh"_a);
+    // There appears to be an issue binding to a static const member of a templated type, so for now
+    // we just use the values constants
+    cls.attr("ASSIGN_STRAYFLUX") = py::cast(0x1);
+    cls.attr("STRAYFLUX_TO_POINT_SOURCES_WHEN_NECESSARY") = py::cast(0x2);
+    cls.attr("STRAYFLUX_TO_POINT_SOURCES_ALWAYS") = py::cast(0x4);
+    cls.attr("STRAYFLUX_R_TO_FOOTPRINT") = py::cast(0x8);
+    cls.attr("STRAYFLUX_NEAREST_FOOTPRINT") = py::cast(0x10);
+    cls.attr("STRAYFLUX_TRIM") = py::cast(0x20);
```

Comment:
We should use the enums rather than repeat their values as integers, even if it's more verbose - don't want any chance of this getting out of sync with the C++.